Source/GSThemeDrawing.m

```diff
@@ -816,6 +816,139 @@ - (void) drawStepperCell: (NSCell*)cell
     [self drawStepperDownButton: downRect];
 }
 
+// NSSwitch drawing methods
+
+- (void) drawSwitchBezel: (NSRect)frame
+                forState: (NSControlStateValue)v
+                 enabled: (BOOL)enabled
+{
+  NSBezierPath *p;
+  NSPoint point;
+  CGFloat radius;
+  NSColor *backgroundColor = [NSColor selectedControlColor]; // onColor;
+
+  if (NSControlStateValueOn != v)
+    {
+      backgroundColor = [NSColor windowBackgroundColor]; // offColor;
+    }
+
+  if (enabled == NO)
+    {
+      backgroundColor = [NSColor disabledControlTextColor];
+    }
+  
+  // make smaller than enclosing frame
+  frame = NSInsetRect(frame, 4, 4);
+  radius = frame.size.height / 2.0;
+  point = frame.origin;
+  point.x += radius;
+  point.y += radius - 0.5;
+
+  // Draw initial path to enclose the button...
+  // left half-circle
+  p = [NSBezierPath bezierPath];
+  [p appendBezierPathWithArcWithCenter: point
+				radius: radius
+			    startAngle: 90.0
+			      endAngle: 270.0];
+
+  // line to first point and right halfcircle
+  point.x += frame.size.width - frame.size.height;
```

Comment:
Mixing width and height to set an x coordinate looks wrong. Most likely you want to do something like 

`point.x += frame.size.width - 2 * radius;`

But even there I am not totally sure.

Reply:
This causes some weirdnes. :) 
![weirdness](https://user-images.githubusercontent.com/27943/80203129-aa115b00-85f4-11ea-8d3a-11eae653b4d1.png)

Reply:
I think a change is needed in the knob rendering method....

This change, as it is, causes some weirdness...

![weirdness](https://user-images.githubusercontent.com/27943/80203327-feb4d600-85f4-11ea-8511-327fcc2e295f.png)

Reply:
I used your calculation (with slight modification) in the knob rendering.  looks better and it seems more correct.

Reply:
That is weired, radius is frame.size.height / 2.0 multiplying that value with 2.0 should give  frame.size.height. This should just give the same result. Are you sure you did not mix up things?
But this is just a way to write things differently, it should result in the same appearance. So this is not important.